### DISCUSSION TITLE
How do delete a row using Prisma with two conditions in where?

### DISCUSSION THREAD

**Discussion by skorotkiewicz** [prisma/prisma#8581] [Q&A]
In my project, I need to delete only if two conditions from where match. But Prisma shows me that AND is not supporting with delete().

And how to get the current data after deletion with delete()?
The delete function returns the data from before the deletion.

Currently in the project I search for a given ID using findFirst, delete the row, filter the output using the js filter() function and send the result to the user.

Is it possible to do this with a single query? As my try below? 

```javascript
      const mikr = await prisma.votes.delete({
        where: {
          AND: [{ messageId: id }, { login: user.login }],
        },
        include: {
          Messages: {
            select: {
              votes: true,
            },
          },
        },
      });

      console.log(mikr);
```

**Comment by ryands17** [comment]
@skorotkiewicz 👋 

`delete` only accepts unique fields. As you're using `AND` in this case, you need to replace `delete` with `deleteMany`.

**Reply by skorotkiewicz** [reply]
But I want to delete only one row that matches messageId and login.
MessageId and Login are a unique fields.

**Reply by ryands17** [reply]
Could you share your schema?

**Reply by skorotkiewicz** [reply]
```prisma
model Votes {
  id Int @id @default(autoincrement())
  Messages  Messages? @relation(fields: [messageId], references: [id])
  messageId Int
  User  User?  @relation(fields: [login], references: [login])
  login String @db.VarChar(255)
}

model User {
  id      String  @id
  login   String  @unique @db.VarChar(255)
  votes             Votes[]
}
```

**Reply by ryands17** [reply]
In this case, they are separately unique, so Prisma will accept only a single value. For passing both the values, you need to add a compound unique index like this:

```prisma
model User {
  id      String  @id
  login   String  @unique @db.VarChar(255)
  votes             Votes[]

  @@unique([id, login])
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
